addons/binding/org.openhab.binding.networkcamera/src/main/java/org/openhab/binding/networkcamera/internal/ftp/FtpServer.java

```diff
@@ -0,0 +1,536 @@
+
+        if (authenticationData.containsKey(username)) {
+            throw new RuntimeException("Crerentials for user '" + username + "' already exists!");
+        }
+
+        authenticationData.put(username, new UsernamePassword(username, password));
+    }
+
+    public synchronized void removeAuthenticationCredentials(String username) throws RuntimeException {
+
+        authenticationData.remove(username);
+    }
+
+    public synchronized void removeEventListener(FtpServerEventListener listener) {
+        listeners.remove(listener);
+    }
+
+    void sendMsgToListeners(String userName, byte[] data) {
+        try {
+            Iterator<FtpServerEventListener> iterator = listeners.iterator();
+
+            while (iterator.hasNext()) {
+                iterator.next().fileReceived(userName, data);
+            }
+
+        } catch (Exception e) {
+            logger.debug("Event listener invoking error: ", e.getMessage());
+        }
+    }
+
+    public void printStats() {
+        FtpStatistics ftpStats = myFTPLet.getStats();
+
+        logger.debug("TotalConnectionNumber: {}", ftpStats.getTotalConnectionNumber());
+        logger.debug("TotalLoginNumber: {}", ftpStats.getTotalLoginNumber());
+        logger.debug("TotalFailedLoginNumber: {}", ftpStats.getTotalFailedLoginNumber());
+        logger.debug("TotalUploadNumber: {}", ftpStats.getTotalUploadNumber());
+        logger.debug("TotalUploadSize: {}", ftpStats.getTotalUploadSize());
+
+        logger.debug("CurrentConnectionNumber: {}", ftpStats.getCurrentConnectionNumber());
+        logger.debug("CurrentLoginNumber: {}", ftpStats.getCurrentLoginNumber());
+    }
+
+    private void initServer() {
+        logger.info("Starting FTP server, port={}, idleTimeout={}", port, idleTimeout);
+
+        FtpServerFactory serverFactory = new FtpServerFactory();
+        ListenerFactory listenerFactory = new ListenerFactory();
+        listenerFactory.setPort(port);
+        listenerFactory.setIdleTimeout(idleTimeout);
+
+        Listener listener = listenerFactory.createListener();
+
+        serverFactory.addListener("default", listener);
+
+        Map<String, Ftplet> ftplets = new LinkedHashMap<String, Ftplet>();
+        myFTPLet = new MyFTPLet();
+
+        ftplets.put("ftplet", myFTPLet);
+
+        serverFactory.setFtplets(ftplets);
+        serverFactory.setFileSystem(new FileSystemFactory() {
+            @Override
+            public FileSystemView createFileSystemView(User user) throws FtpException {
+                logger.debug("createFileSystemView: {}", user.getName());
+                return new MyFileSystemView();
+            }
+        });
+
+        // set the user manager
+        serverFactory.setUserManager(new FTPUserManager());
+        server = serverFactory.createServer();
+        try {
+            server.start();
+        } catch (FtpException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private class MyFTPLet extends DefaultFtplet {
```

Comment:
Usually one or two small private classes fit nicely into a class, but I think you should factor out yours into extra files because they are too many which makes it hard to read.